pkg/redirectpolicy/redirectpolicy.go

```diff
@@ -0,0 +1,173 @@
+//  Copyright 2020 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package redirectpolicy
+
+import (
+	"fmt"
+	"github.com/cilium/cilium/pkg/k8s"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	k8sUtils "github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"k8s.io/apimachinery/pkg/types"
+	"net"
+	"strconv"
+)
+
+type frontend = loadbalancer.L3n4Addr
+
+type backend = loadbalancer.L3n4Addr
+
+// Policy ID that includes its name and namespace
+type policyID = string
+
+func (config LRPConfig) getPolicyID() policyID {
+	return config.name + config.namespace
+}
+
```

Comment:
Nit: we can move these to the bottom of the file as the order of declarations / definitions should be most-important to least-important. Here `Parse` is the piece that most developers are interested in and should be the first one they see.